### PR TITLE
Add some tests for prioq.

### PR DESCRIPTION
**Pull request by schmonz**

### CODE REVIEW

**Review by mbhangui — APPROVED** [review]

File: tests/unittest_prioq.c [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    pe.dt = priority;
+    pe.id = value;
+    ck_assert_int_ne(0, prioq_insert(&pq,&pe));
+  }
+
+  value = 0;
+  while (prioq_min(&pq,&pe)) {
+    prioq_delmin(&pq);
+    ck_assert_uint_eq(value, pe.id);
+    value++;
```
Comment (DerDakon) [inline tests/unittest_prioq.c:61]: check and poison pe.dt here. Same below for most other tests, too.

File: tests/unittest_prioq.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  unsigned long value;
+
+  for (value = 0; value < 5; value++) {
+    long priority = 12345 + value;
+    pe.dt = priority;
+    pe.id = value;
+    ck_assert_int_ne(0, prioq_insert(&pq,&pe));
+  }
+
+  value = 0;
```
Comment (DerDakon) [inline tests/unittest_prioq.c:57]: poison pe.dt here

File: tests/unittest_prioq.c [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    sorted[i] = i;
+  }
+
+  i = 0;
+  while (prioq_min(&pq,&pe)) {
+    prioq_delmin(&pq);
+    actual[i] = pe.id;
+    i++;
+  }
+
```
Comment (DerDakon) [inline tests/unittest_prioq.c:120]: check i == 5 out of paranoia here.

**Review by DerDakon — CHANGES_REQUESTED** [review]
Be a bit more paranoid about reused values ;) LGTM otherwise.

**Review by DerDakon — APPROVED** [review]

**Review by josuah — APPROVED** [review]
Tests are a good way to document libraries through providing examples.

Do we aim covering more of the code base this way?

Thank you and ok josuah

**Comment by josuah** [comment]
No patch broke:

master https://notqmail.z0.is/commit/0b1896650c2ba7100df450d698752090e6ba3b22/
this https://notqmail.z0.is/commit/acf8ff62ecee706bd61185712a21e91337ace76b/

**Comment by schmonz** [comment]
> Tests are a good way to document libraries through providing examples.
> 
> Do we aim covering more of the code base this way?

I personally enjoy it, which is why I wrote these particular tests. Conveniently, getting code under automated test is a huge lever for preserving and improving our code -- and therefore ultimately for making notqmail development faster and more fun. 😀

**Pull request merged**